cmd/virtual-workspaces/main.go

```diff
@@ -33,17 +33,20 @@ import (
 )
 
 func main() {
-	stopCh := genericapiserver.SetupSignalHandler()
+	//signal handler
+	ctx := genericapiserver.SetupSignalContext()
 
 	rand.Seed(time.Now().UTC().UnixNano())
 
 	pflag.CommandLine.AddGoFlagSet(goflag.CommandLine)
 
-	command := NewVirtualWorkspaceApiServerCommand(stopCh)
+	command := NewVirtualWorkspaceApiServerCommand(ctx.Done())
 	if err := command.Execute(); err != nil {
 		fmt.Fprintf(os.Stderr, "%v\n", err)
 		os.Exit(1)
 	}
+
+	<-ctx.Done()
```

Comment:
why is this context wait necessary?

Reply:
in cluster-controller it is needed because there everything runs in go routines in the background. Not here.

Reply:
ohh sorry, will change that then